test/sql/timezone/test_icu_timezone.test

```diff
@@ -18,6 +18,16 @@ TimeZone	America/Los_Angeles	The current time zone	VARCHAR
 statement error
 SET TimeZone = 'fnord';
 
+query I
+SELECT  '2001-02-16 20:38:40'::TIMESTAMP AT TIME ZONE 'America/Denver';
+----
+2001-02-16 19:38:40-08
+
+query I
+SELECT '2001-02-16 20:38:40-05'::TIMESTAMPTZ AT TIME ZONE 'America/Denver';
```

Comment:
This function seems to be relatively complex but does not have that many tests, could we add tests with:

* NULL as time zone
* NULL as timestamp
* Infinity timestamps
* Non-existent time zone
* All of the above but with timestamps/time zones stored in a table (i.e. not constant)

Reply:
Yeah sorry I just copied and pasted a bunch of existing code. Coming shortly.